### ISSUE TITLE
Reload window creates new task

### ISSUE THREAD

**Issue by weinand** [microsoft/vscode#158886]
Testing #158787

- open vscode workspace in VS Code
- F1 -> "Run Task" -> "Run Code Server"

Observe: a task show up:
<img width="945" alt="CleanShot 2022-08-23 at 14 09 54@2x" src="https://user-images.githubusercontent.com/1898161/186154515-6ae07214-fd80-4989-9fdc-c7568bff55fa.png">

- F1 -> "Reload Window"

Observe: another "Run Code Server" shows up:
<img width="952" alt="CleanShot 2022-08-23 at 14 11 05@2x" src="https://user-images.githubusercontent.com/1898161/186154697-885ba659-d6aa-4c03-ae31-ce263a5dc59a.png">

**Comment by meganrogge** [comment]
This should not be happening, but also don't believe this kind of task should be persisted, so will be fixed when #158790 is

**Comment by meganrogge** [comment]
were you able to reproduce the issue with a build task?

**Comment by weinand** [comment]
@meganrogge no, the issue only occurred for the "Run Code Server" task. The build task and all the others I tried worked fine.

**Comment by meganrogge** [comment]
likely /closedWith 95e0f54ec0ff766726e4e1e5b20a3d0b2f33bdc0

**Comment by andreamah** [comment]
Seems that it's still happening:

<img width="913" alt="image" src="https://user-images.githubusercontent.com/31675041/186769920-55fa0cf8-a561-45b7-b466-5a9ecc2ba213.png">

**Comment by meganrogge** [comment]
the persistent task's map key and the one before reload are diff't 🤔

**Comment by meganrogge** [comment]
`getCommonTaskId` is what I should be using I think. that fixes this issue
